modules/mockk-agent/src/jvmMain/kotlin/io/mockk/proxy/jvm/ObjenesisInstantiator.kt

```diff
@@ -24,6 +24,11 @@ class ObjenesisInstantiator(
         if (cls == Any::class.java) {
             @Suppress("UNCHECKED_CAST")
             return Any() as T
+        } else if (cls.isSealedSafe()) {
+            @Suppress("UNCHECKED_CAST")
+            return cls.getPermittedSubclassesSafe().firstNotNullOfOrNull { subCls ->
```

Comment:
@aSemy: Your PR did not have the `return` here and IIRC the mockk tests were also running successfully without. Was that an oversight or am I missing something?

Reply:
Good spot! Yes that's a mistake, I intended to use a `return`. However, it does show that Objenesis can probably handle sealed classes and we don't need to have custom code.

```kotlin
    private val instantiators = Collections.synchronizedMap(WeakHashMap<Class<*>, ObjectInstantiator<*>>())
    override fun <T> instance(cls: Class<T>): T {
        if (cls == Any::class.java) {
            @Suppress("UNCHECKED_CAST")
            return Any() as T
        } else if (!Modifier.isFinal(cls.modifiers) && !cls.isSealedSafe()) {
            try {
                val instance = instantiateViaProxy(cls)
                if (instance != null) {
                    return instance
                }
            } catch (ex: Exception) {
                log.trace(
                    ex, "Failed to instantiate via proxy " + cls + ". " +
                            "Doing objenesis instantiation"
                )
            }
        }
        return instanceViaObjenesis(cls)
    }
```

Reply:
Thought so as well - and the mockk tests are indeed running successfully. But in our code that leads to the original `InstantiationError`:

```
java.lang.InstantiationError: org.example.SealedClass
        at jdk.internal.reflect.GeneratedSerializationConstructorAccessor7.newInstance(Unknown Source)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
        at org.objenesis.instantiator.sun.SunReflectionFactoryInstantiator.newInstance(SunReflectionFactoryInstantiator.java:48)
        at io.mockk.proxy.jvm.ObjenesisInstantiator.instanceViaObjenesis(ObjenesisInstantiator.kt:83)
        at io.mockk.proxy.jvm.ObjenesisInstantiator.instance(ObjenesisInstantiator.kt:42)
        at io.mockk.impl.instantiation.JvmInstantiator$instantiate$2.invoke(JvmInstantiator.kt:16)
```
I don't yet have a theory _why_ but I can see that our code is coming from the subclass creation.

But it looks like actually the changes in `ProxyMaker` are enough to get our tests running. I am still struggling to create a corresponding test case in mockk, though. Will update this PR later.